randomizer.py

```diff
@@ -1152,6 +1156,169 @@ def mutate(self):
+            self.formula = 2
+            self.inflict_status = 0
+
+        if self.formula == 2:
+            # Formula 2 calls the "inflict status" value as a spell to cast 25% of the time
+            if (value == 0) or (randint(1,10) == 1):
+                # 10% chance for pre-existing spell casts to be randomized
+                # Value is capped at FF internally, so no abilities past Holy Bracelet
+                newvalue = randint(1,0xFF)
+                if newvalue in [0x28, 0x2D, 0xB8, 0xDB, 0xDC]:
+                    # Empty abilities
+                    newvalue = randint(1,0x1F)
+                self.inflict_status = newvalue
+
+    def mutate_status(self):
+        if (not (self.formula == 2)) and randint(1,10) == 1:
+            value = self.inflict_status
+            if value > 0 and randint(1,10) != 1:
+                # 1% Chance for a pre-existing Inflict Status to be randomized; 10% otherwise
+                return
+            newvalue = randint(0,0x79)
+            if newvalue == 0x60:
+                # Banning Crystal (since it's more likely to appear on weapons)
+                return
+            self.inflict_status = newvalue
+
+
+class ShieldObject(TableObject):
+    def mutate(self):
+        for attr in ["physical_evade", "magic_evade"]:
+            value = getattr(self, attr)
+            if random.choice([True, False]) and 0 <= value <= 0x50:
+                    newvalue = mutate_normal(value, minimum=0, maximum=0x50) # Max 80
+                    setattr(self, attr, newvalue)
+
+
+class ArmorObject(TableObject):
+    def mutate(self):
+        for attr in ["hp_bonus", "mp_bonus"]:
+            value = getattr(self, attr)
+            if random.choice([True, False]) and 0 <= value <= 0xFD:
+                    newvalue = mutate_normal(value, minimum=0, maximum=0xFD)
+                    setattr(self, attr, newvalue)
+
+
+class AccessoryObject(TableObject):
+    def mutate(self):
+        for attr in ["physical_evade", "magic_evade"]:
+            value = getattr(self, attr)
+            if random.choice([True, False]) and 0 <= value <= 0x3C:
+                    newvalue = mutate_normal(value, minimum=0, maximum=0x3C) # Max 60
+                    setattr(self, attr, newvalue)
+
+
+class ChemistItemObject(TableObject):
+    def mutate(self):
+        value = self.zval
+        if 1 <= value <= 0xFD:
+            newvalue = mutate_normal(value, minimum=1, maximum=0xFD)
+            self.zval = newvalue
+
+
+class InflictStatusObject(TableObject):
+    def randomize_empty(self):
+        if (0x1D <= self.index <= 0x1F) or  (0x7A <= self.index <= 0x7F):
+            toinflict = 0
+            while True:
+                toinflict |= (1 << randint(0, 39))
+                toinflict &= VALID_START_STATUSES
+                if (toinflict and
+                        randint(1, 2**(bin(toinflict).count("1"))) != 1):
+                    break
```

Comment:
Scratch that. The odds of additional effects are actually significantly lower, because not all status effects are valid. I'll try to fix that before I merge this branch.